src/routers/transactions/money/loans.ts

```diff
@@ -1,24 +1,194 @@
+ * @apiSuccessExample Success-Response:
+   {
+     "uuid": "9f11efb5-d5b4-4853-aa88-ee10c2940c9f",
+   }
+ */
 router.post('/', async (req, res) => {
-  // TODO
-  res.status(StatusCode.CREATED).send('Successfully retrieved new loan');
-}, authorized(UserType.ADMIN));
+  const createReq: LoanCreationReq = {
+    uuid: req.body.uuid,
+    dueDate: req.body.dueDate,
+    datetime: req.body.datetime,
+    toPersonUuid: req.body.toPersonUuid,
+    fromPersonUuid: req.body.fromPersonUuid,
+    amount: req.body.amount,
+    currency: req.body.currency,
+  };
+
+  try {
+    const uuid = await LoansService.createLoanInDb(createReq);
+    res.status(StatusCode.CREATED).send({ uuid });
+  } catch (e) {
+    if (e.message === LoansService.unhandledErrorMsg) {
+      res.status(StatusCode.INTERNAL_SERVER_ERROR).send();
+    } else {
+      res.status(StatusCode.BAD_REQUEST).send('BadRequest ' + e.message);
+    }
+  }
+}, authorized(UserType.ADMIN, UserType.TRADER));
 
+/**
+ * @api {put} /transactions/money/loans/:uuid
+ * @apiName UpdateLoan
+ * @apiGroup Loans
+ * @apiVersion 0.0.1
+ * @apiDescription Updates a loan entry.
+ *              Any fields not given will not be updated.
+ *              Extra non existent fields will not be ignored.
+ *              Returns the loan's actual data on error or success.
+ *
+ * @apiParam {String} uuid The UUID of the loan to update.
+ * @apiParam {String} dueDate The date that the loan should be repaid by.
+ * @apiParam {String} datetime The date that the loan was given.
+ * @apiParam {String} toPersonUuid The person receiving the loan.
+ * @apiParam {String} fromPersonUuid The person giving the loan.
+ * @apiParam {Number} amount The amount of money exchanged.
+ * @apiParam {String} currency The currency type of the amount exchanged, such as UGX.
+ *
+ * @apiError (403) Forbidden Current user type does not have sufficient privileges.
+ * @apiError (404) NotFound Product export not found
+ *
+ * @apiSuccess (200) {String} Successfully updated loan entry
+   {
+    "uuid": "7359efb5-d5b4-4853-aa88-ee10c2940ab3",
+    "dueDate":      "2018-01-23 04:06:06.000Z",
+    "datetime":      "2018-01-23 04:05:06.000Z",
+    "toPersonUuid": "2648e3a5-a88d-473b-9d4a-74a2153970a4",
+    "fromPersonUuid": "dc25a3a5-a99d-473b-9d4a-74a21539ff24",
+    "amount": 10.22
+    "currency": "UGX"
+    "toPersonName": "Jerry Morton",
+    "fromPersonName": "Joey Goold"
+   }
+*/
 router.put('/:uuid', async (req, res) => {
-  // TODO
-  res.status(StatusCode.OK).send('Successfully retrieved updated loan');
-}, authorized(UserType.ADMIN));
+  const updateReq: LoanUpdateReq = {
+    uuid: req.params.uuid,
+
+    // optional parameters
+    dueDate: req.body.dueDate,
+    datetime: req.body.datetime,
+    toPersonUuid: req.body.toPersonUuid,
+    fromPersonUuid: req.body.fromPersonUuid,
+    amount: req.body.amount,
+    currency: req.body.currency,
+  };
+
+  try {
+    await LoansService.updateLoanInDb(updateReq);
```

Comment:
instead of doing an update and then a get, is it possible to do an update and it returns the updated element? kind of like how the insert functionality will return the inserted data after

Reply:
Yea, but loans will need to aggregate the return from both the money and loan db call, if you really want me to do it I will. But I think since this is the last change to the core, not worth the effort.